idaes/core/util/scaling.py

```diff
@@ -49,12 +50,36 @@
 from pyomo.core import expr as EXPR
 from pyomo.common.numeric_types import native_types
 from pyomo.core.base.units_container import _PyomoUnit
+from pyomo.common.fileutils import find_library
 
 import idaes.logger as idaeslog
 
 _log = idaeslog.getLogger(__name__)
 
 
+# There appears to be a bug in the ASL which causes terminal failures
+# if you try to create multiple ASL structs with different external
+# functions in the same process. This causes pytest to crash during testing.
+# To avoid this, register all known external functions before we call
+# PyNumero.
+ext_funcs = ["cubic_roots", "general_helmholtz_external", "functions"]
+library_set = set()
+libraries = []
+
+for f in ext_funcs:
+    library = find_library(f)
```

Comment:
This relies on IDAES having modified the path, right? Would it make sense to be more explicit about the libraries we are adding? I.e. use `ideas.bin_directory`. The idea is just that maybe the user has other copies of these libraries on the path (or others with the same names, `functions` in particular is pretty generic).

Reply:
I am not sure there - `find_library` is a Pyomo function so I am not sure what it really does. I am not really aware of the `idaes.bin_directory` either - I assume it is the path to where the binaries are so we could just use `path.join(idaes.bin_directory, "function_name.so")`?

Reply:
@Robbybp I am not sure why, but it looks like there is a differnce in behaviour on Windows and Linux if I try this. On Linux, using `path.join(idaes.bin_directory, "function_name.so")` appears to work, but on Windows we see the same terminal failure during testing due to the ASL bug.

For now, I think I will stick with using Pyomo's `find_library` function as it appear to work for both OSes.

Reply:
This is probably because the libraries have the `.dll` extension on Windows (and the `.dylib` extension on Mac). I would prefer to be more specific about the libraries we're adding to minimize the chance that something unexpected happens. Maybe we could pass `path.join(ideas.bin_directory, libname)` to `find_library`? Here we're only using `find_library` to deal with the OS-specific extension, but I think that's appropriate.

Reply:
@Robbybp It looks like that is working now.